File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
from collections import Counter


class Solution:
    def firstUniqChar(self, s: str) -> int:
        sCounter = Counter(s)
        for idx, char in enumerate(s):
            if sCounter[char] == 1:
                return idx
        return -1


s = "loveleetcode"
print(Solution().firstUniqChar(s))

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
 public:
  int firstUniqChar(string s) {
    int count[26] = {0};
    int n = s.length();
    for (int i = 0; i < n; i++) {
      int index = s[i] - 'a';
      count[index]++;
    }
    for (int i = 0; i < n; i++) {
      int index = s[i] - 'a';
      if (count[index] == 1) {
        return i;
      }
    }
    return -1;
  }
};